pkg/controller/rollout/rollout_status.go

```diff
@@ -349,7 +349,14 @@ func (r *RolloutReconciler) handleFinalizer(rollout *v1beta1.Rollout) error {
 
 func getRolloutID(workload *util.Workload) string {
 	if workload != nil {
-		return workload.Labels[v1beta1.RolloutIDLabel]
+		rolloutID := workload.Labels[v1beta1.RolloutIDLabel]
+		if rolloutID == "" {
+			rolloutID = workload.CanaryRevision
+		}
+		if workload.IsInRollback {
```

Comment:
it is not necessary to add prefix to rollout id if the user had explicitly specify rollout id in the workload label. Consider only add prefix if label in rollout is empty.